src/LayerEditor/ui/mainwindow.py

```diff
@@ -322,12 +323,38 @@ def update_recent_files(self, from_row=1):
     #     #   self.diagramScene.init_area.reload()
     #     pass
     #
-    # def closeEvent(self, event):
-    #     """Performs actions before app is closed."""
-    #     # prompt user to save changes (if any)
-    #     if not self._layer_editor.save_old_file():
-    #         return event.ignore()
-    #     super(MainWindow, self).closeEvent(event)
+    def closeEvent(self, event):
+        """Performs actions before app is closed."""
+        # prompt user to save changes (if any)
+        if not self._layer_editor.save_old_file():
+            return event.ignore()
+        super(MainWindow, self).closeEvent(event)
+
+    def undo(self):
+        self._layer_editor.le_data.gui_curr_block.selection.deselect_all()
+        # Deselect because selected region can change and that could create wrong behaviour #
+        self.curr_scene.hide_aux_point_and_seg()
+        undo.stack().undo()
+        self.curr_scene.update_scene()
```

Comment:
Rather introduce an update method, we should also update layer_panel and surface later on.

Reply:
Indeed, i noticed that as well and method `update_all` was created.